content/blog/2020/01/2020-01-09-vscode-caseStudy.adoc

```diff
@@ -0,0 +1,71 @@
+---
+layout: post
+title: "Validating JCasC configuration files using Visual Studio Code"
+opengraph:
+  image: /images/logos/JCasC/JCasC.png
+tags:
+- community-bridge
+- JCasC
+- VSCode
+- community
+author: sladyn98
+---
+
+
+==== Configuration-as-code plugin
+The link:https://plugins.jenkins.io/configuration-as-code[Configuration as Code plugin] has been designed as an opinionated way to configure Jenkins based on human-readable declarative configuration files. Writing such a file should be feasible without being a Jenkins expert, just translating into code a configuration process one is used to executing in the web UI. The plugin uses a schema to verify the files being applied to the Jenkins instance.
+
+With the new JSON Schema being enabled developers can now test their yaml file against it. The schema checks the `descriptors` i.e. configuration that can be applied to a plugin or Jenkins core, the correct type is used and help text is provided in some cases. VSCode allows us to test out the schema right out of the box with some modifications.
+
+=== Steps to Enable the Schema Validation
+
+a) The first step includes installing the YAML plugin from Red Hat. Open up the Extension list in VSCode editor using `Ctrl + Shift + X`.
```

Comment:
shouldn't you install the jcasc vs code extension first? and then the other one will be suggested?

Reply:
Also YAML plugin is installed for you because of the dependency on it 🤔

Reply:
Yeah correct, we install the VSCode Extension and the other one is installed automatically.